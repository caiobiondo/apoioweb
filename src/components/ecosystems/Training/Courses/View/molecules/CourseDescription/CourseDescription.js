import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-flexbox-grid';
import { TittleWrapper, Divider, DescriptionWrapper } from './CourseDescription.styles';

class CourseDescription extends Component {
  render() {
    const { course } = this.props;

    return (
      <div>
        <Row>
          <Col md={12} sm={12}>
            <TittleWrapper>{course.title}</TittleWrapper>
          </Col>
        </Row>
        <Divider />
        <Row>
          <Col md={12} sm={12}>
            <DescriptionWrapper>{course.description}</DescriptionWrapper>
          </Col>
        </Row>
      </div>
    );
  }
}

CourseDescription.propTypes = {
  course: PropTypes.object.isRequired,
};

export default CourseDescription;
