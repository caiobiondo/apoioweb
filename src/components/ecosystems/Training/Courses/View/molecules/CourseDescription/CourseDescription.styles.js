import styled from 'styled-components';
import { fs13, fs14, fs20, fs32, fw700, NaturaBold, RobotoRegular } from 'styles/typography';
import { gray890 } from 'styles/colors';
import { Responsive, getTheme } from '@entria/components';

export const TittleWrapper = styled.div`
  font-size: ${fs32};
  font-family: ${NaturaBold};
  font-weight: ${fw700};

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    font-size: ${fs20};
  }
`;

export const Divider = styled.span`
  &:after {
    width: 35px;
    height: 3px;
    background: ${getTheme().palette.primary1Color};
    content: '';
    position: absolute;
    margin-top: 10px;
    border-radius: 5px;
  }
`;

export const DescriptionWrapper = styled.p`
  font-size: ${fs14};
  font-family: ${RobotoRegular};
  padding: 15px 0;
  text-align: left;
  line-height: 1.5;
  color: ${gray890};
`;
