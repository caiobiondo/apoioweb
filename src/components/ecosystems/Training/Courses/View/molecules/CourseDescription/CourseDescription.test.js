import React from 'react';
import { render } from 'enzyme';
import { CourseDescription } from './CourseDescription';

describe('CourseDescription', () => {
  it('should render course description and title', () => {
    // given
    const props = {
      course: {
        id: 1,
        title: 'Title',
        description: 'Description',
      },
      sellerId: 123,
    };

    // when
    const result = render(<CourseDescription {...props} />);

    // then
    expect(result).toMatchSnapshot();
  });
});
