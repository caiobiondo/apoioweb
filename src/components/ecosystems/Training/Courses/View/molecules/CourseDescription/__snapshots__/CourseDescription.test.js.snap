// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseDescription should render course description and title 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-md-12 col-sm-12"
    >
      <div
        class="CourseDescriptionstyles__TittleWrapper-eqWawD cHJOJC"
      >
        Title
      </div>
    </div>
  </div>
  <span
    class="CourseDescriptionstyles__Divider-eXAsJP hbbiom"
  >
    
  </span>
  <div
    class="row"
  >
    <div
      class="col-md-12 col-sm-12"
    >
      <p
        class="CourseDescriptionstyles__DescriptionWrapper-fnSggu jdkzkp"
      >
        Description
      </p>
    </div>
  </div>
</div>
`;
